submissions/iva-stasia/frogger-game/app.js

```diff
@@ -0,0 +1,184 @@
+"use strict";
+
+const INIT_VAL = {
+    playerX: 202,
+    playerY: 410,
+    stepX: 101,
+    stepY: 85,
+    enemyX: -101,
+    enemyY: [63, 147, 230],
+    enemyMinSpeed: 100,
+    enemyMaxspeed: 200,
+    collision: 75,
+    canvasWidth: 505,
+    canvasHeight: 606,
+};
+
+const allPlayerSprites = [
+    'images/char-boy.png',
+    'images/char-cat-girl.png',
+    'images/char-princess-girl.png'
+];
+
+let levelSpeed = 0;
+let level = 0;
+
+/*---------- DOM elements ----------*/
+
+const body = document.body;
+
+body.insertAdjacentHTML('afterbegin',
+    `
+        <div class='level'>Level: ${level}/5</div>
+        <div class='win_message'><h2>You won :)</h2><span>press enter</span></div>
+        <div class='lost_message'><h2>You lost :(</h2><span>press enter</span></div>
+        <div class='player_option'></div>
+    `);
+    
+const levelDiv = document.querySelector('.level');
+const winMessage = document.querySelector('.win_message');
+const lostMessage = document.querySelector('.lost_message');
+const playerOptionDiv = document.querySelector('.player_option');
+
+playerOptionDiv.innerHTML =
+    `
+        <h1>Choose your character</h1>
+        <img src='images/char-boy.png' class='player_img' data-player-num = '0' />
+        <img src='images/char-cat-girl.png' class='player_img' data-player-num = '1' />
+        <img src='images/char-princess-girl.png' class='player_img' data-player-num = '2' />
+    `;
+
+/*---------- Enemy ----------*/
+
+const Enemy = function(x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x += this.speed * dt + levelSpeed;
+
+    if (this.x > INIT_VAL.canvasWidth
+        || !playerOptionDiv.classList.contains('hide')
+        || lostMessage.classList.contains('active_message')
+        || winMessage.classList.contains('active_message')) {
+        this.x = INIT_VAL.enemyX;
+    };
+
+    this.checkCollision();
+};
+
+Enemy.prototype.checkCollision = function() {
+    if (player.x < this.x + INIT_VAL.collision &&
+        player.x + INIT_VAL.collision > this.x &&
+        player.y < this.y + INIT_VAL.collision &&
```

Comment:
Please find a requirement among those posted/linked by bot that addresses this case, post it as a comment and also fix the code accordingly.

Find other fragments where the same requirement is not met and fix them as well.

Reply:
Sorry, I'm not sure I understood correctly.
Do you mean this?

> Classes do not refer to any global variables, like global variable player, which is an instance of Player class (referring to global constants and globals provided by the gaming platform like Resources is OK); Hint: pass instance of a game object (or objects) as an argument to other game objects they need to interact with.

If so, at the moment, I don't understand how I can check for enemy and player collision without specifying either one or the other. Even with a hint :(
Can I create a separate function that does not inherit anything?
I will try to understand what the hint means or ask the students for help.

Or does it mean that I can pass the player as an argument? In this way:
<pre><code>Enemy.prototype.checkCollision = function(player) {}
</code></pre>

Reply:
:white_check_mark:

If this solution doesn't work, I have another one. But it was suggested to me in the students' chat, so first I want to try it my way.

Reply:
You still refer to a global variable within class methods definition. Just in different place.
The requirements offer a hint on how to resolve this issue.

Reply:
I hope I finally figured it out. Also added calculation of initial positions of enemies.